app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt

```diff
@@ -75,13 +79,31 @@ class MoveFragment : MenuFoldersFragment() {
         super.onViewCreated(view, savedInstanceState)
         setSystemBarsColors()
 
+        bindAlertToViewLifecycle(inputDialog)
+        setupRecyclerView()
         setupListeners()
-        setupFolderAdapters()
+        observeFolders()
         setupCreateFolderDialog()
         observeNewFolderCreation()
         observeSearchResults()
```

Comment:
We can probably regroup the `observeFolders()` and the `observeSearchResults()` inside a single liveData if you use a single RecyclerView. This would also remove the need for the `isSearching` global boolean and make everything more robust

Reply:
We can't.
`observeFolders()` triggers only once, and `observeSearchResults()` triggers every time a search is done.

Reply:
Fixed in https://github.com/Infomaniak/android-kMail/pull/1901

Reply:
The linked PR doesn't really solve the issue from my point of view, maybe we can talk about it. Also, is there a specific reason to delay the fix to a future PR instead of including it in this one? If possible, it might be more efficient to resolve issues in the current PR to avoid tracking a complication like this through multiple PRs